src/range.cr

```diff
@@ -207,11 +256,16 @@ struct Range(B, E)
   # (1...10).includes?(10) # => false
   # ```
   def includes?(value)
-    if @exclusive
-      @begin <= value < @end
-    else
-      @begin <= value <= @end
-    end
+    begin_value = @begin
+    end_value = @end
+
+    # TOOD: change to `nil?` after removing the `nil?` error related to `Pointer`
```

Comment:
```suggestion
    # TODO: change to `nil?` after removing the `nil?` error related to `Pointer`
```

Reply:
Also, which error?

Reply:
@bcardif #7180

Reply:
Thanks, fixed.

This is the error: https://play.crystal-lang.org/#/r/5xke

It was an old error because at some pointer you would use `nil?` to check for a null pointer, but this was later changed to `null?` and the error was introduced to avoid silent breaking changes.

I already removed it in master (#7180) but of course the current compiler would still complain.